packages/jest-cli/src/plugins/quit.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+import WatchPlugin from '../watch_plugin';
+
+class QuitPlugin extends WatchPlugin {
+  showPrompt(): Promise<void> {
+    this._stdout.write('\n');
+    process.exit(0);
+    return Promise.resolve();
```

Comment:
can just mark the function as `async` and drop the `return`, although the transpiled code is pretty ugly. Up to to you 🙂 

(would be cool if babel just wrapped the return value in `Promise.resolve` if there were no `await`s in an `async function`)